.buildkite/pipeline-utils/ci-stats/pick_test_group_run_order.ts

```diff
@@ -273,7 +275,9 @@ export async function pickTestGroupRunOrder() {
           ]
         : []),
       // if we are running on a external job, like kibana-code-coverage-main, try finding times that are specific to that job
-      ...(!prNumber && pipelineSlug !== 'kibana-on-merge'
+      // also skipping triggered jobs that can be running tests against older commits
+      // which may reflect out of date build times when running against the branch head
+      ...(!prNumber && pipelineSlug !== 'kibana-on-merge' && pipelineSource !== 'trigger_job'
```

Comment:
This is not a good fix.  More accurate would be a comparison between the branch head and `BUILDKITE_COMMIT`, need to investigate more.

Reply:
https://github.com/elastic/kibana/pull/170574#issuecomment-1793244806